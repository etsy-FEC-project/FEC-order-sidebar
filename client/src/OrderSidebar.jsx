import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import OrderForm from './OrderForm';
import Overview from './Overview';
import Shipping from './Shipping';
import Favorite from './Favorite';

class OrderSidebar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: this.props.data,
    };
  }

  componentDidMount() {
    fetch(`http://127.0.0.1:541${location.pathname}data`)
      .then(response => response.json())
      .then(data => this.setState({ data }))
      .catch(error => console.error(error));
  }

  render() {
    return (
      <div id="container">
        <OrderForm data={this.state.data.orderForm} />
        <Overview data={this.state.data.overview} />
        <Shipping data={this.state.data.shipping} />
        <Favorite />
      </div>
    );
  }
}

OrderSidebar.defaultProps = {
  data: {
    orderForm: {
      title: 'Star Trek - Captains Oath - Mission Oath - Quote Typography Art Poster Print - (Available in Many Sizes)',
      sellerName: 'HarknettPrints',
      contactName: 'Jamie [Harknett]',
      variationTypes: ['Size'],
      variations: {
        Size: [['6x4 inches', 7.29], ['5x7 inches', 8.84], ['8x10 inches', 14.74]],
      },
      quantity: 29,
    },
    overview: {
      materials: ['Satin Photo Card', 'Photoshop'],
      isHandmade: true,
      isProduct: true,
      whenMade: 'To order',
      numReviews: 1995,
      numFavorites: 2573,
      acceptGiftCards: true,
    },
    shipping: {
      timeToShip: '1-2 business days',
      shipOrigin: 'United Kingdom',
      acceptReturn: true,
      acceptExchange: true,
      acceptCancel: false,
    },
  },
};

OrderSidebar.propTypes = {
  data: PropTypes.shape({
    orderForm: PropTypes.shape({
      title: PropTypes.string.isRequired,
      sellerName: PropTypes.string.isRequired,
      contactName: PropTypes.string.isRequired,
      variationTypes: PropTypes.arrayOf(PropTypes.string),
      variations: PropTypes.objectOf(PropTypes.array),
      quantity: PropTypes.number,
      // numInCarts: PropTypes.number, // For special message
    }).isRequired,
    overview: PropTypes.shape({
      materials: PropTypes.arrayOf(PropTypes.string),
      isHandmade: PropTypes.bool.isRequired,
      isProduct: PropTypes.bool.isRequired,
      whenMade: PropTypes.string.isRequired,
      numReviews: PropTypes.number.isRequired,
      numFavorites: PropTypes.number.isRequired,
      acceptGiftCards: PropTypes.bool.isRequired,
    }).isRequired,
    shipping: PropTypes.shape({
      timeToShip: PropTypes.string.isRequired,
      shipOrigin: PropTypes.string.isRequired,
      acceptReturn: PropTypes.bool.isRequired,
      acceptExchange: PropTypes.bool.isRequired,
      acceptCancel: PropTypes.bool.isRequired,
    }).isRequired,
  }),
};

ReactDOM.render(<OrderSidebar />, document.getElementById('order-sidebar'));
