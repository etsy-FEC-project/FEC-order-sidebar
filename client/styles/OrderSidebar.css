.container {
  width: 392px;
  height: 1500px;
}