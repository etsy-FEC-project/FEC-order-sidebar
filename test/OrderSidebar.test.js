import React from 'react';
import OrderSidebar from '../client/src/OrderSidebar.jsx';

describe('orderSidebar', () => {
  const fakeData = {
    orderForm: {
      title: 'Widget',
      sellerName: 'fooBar Unlimited',
      contactName: 'Fred',
      variationTypes: ['Size'],
      variations: { Size: [['S', 10], ['M', 7], ['L', 11]] },
      quantity: 3,
    },
    overview: {
      materials: ['flesh', 'blood'],
      isHandmade: true,
      isProduct: false,
      whenMade: '90s',
      numReviews: 100,
      numFavorites: 200,
      acceptGiftCards: true,
    },
    shipping: {
      timeToShip: '1-2 business days',
      shipOrigin: 'Valhalla',
      acceptReturn: true,
      acceptExchange: true,
      acceptCancel: false,
    },
  };
  global.fetch = jest.fn(() => new Promise(resolve => resolve({ json: () => fakeData })));

  const wrapper = shallow(<OrderSidebar />);

  it('renders four subcomponents with appropriate data prop', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updates state with result of fetch', () => {
    const sidebar = wrapper.instance();
    expect(sidebar.state.data).toEqual(fakeData);
  });
});
