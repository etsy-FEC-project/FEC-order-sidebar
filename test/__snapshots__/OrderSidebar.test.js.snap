// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderSidebar renders four subcomponents with appropriate data prop 1`] = `
<div
  id="container"
>
  <OrderForm
    data={
      Object {
        "contactName": "Jamie [Harknett]",
        "quantity": 29,
        "sellerName": "HarknettPrints",
        "title": "Star Trek - Captains Oath - Mission Oath - Quote Typography Art Poster Print - (Available in Many Sizes)",
        "variationTypes": Array [
          "Size",
        ],
        "variations": Object {
          "Size": Array [
            Array [
              "6x4 inches",
              7.29,
            ],
            Array [
              "5x7 inches",
              8.84,
            ],
            Array [
              "8x10 inches",
              14.74,
            ],
          ],
        },
      }
    }
  />
  <Overview
    data={
      Object {
        "acceptGiftCards": true,
        "isHandmade": true,
        "isProduct": true,
        "materials": Array [
          "Satin Photo Card",
          "Photoshop",
        ],
        "numFavorites": 2573,
        "numReviews": 1995,
        "whenMade": "To order",
      }
    }
  />
  <Shipping
    data={
      Object {
        "acceptCancel": false,
        "acceptExchange": true,
        "acceptReturn": true,
        "shipOrigin": "United Kingdom",
        "timeToShip": "1-2 business days",
      }
    }
  />
  <Favorite />
</div>
`;
